rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -192,10 +192,11 @@ error_code sys_fs_open(vm::cptr<char> path, s32 flags, vm::ptr<u32> fd, s32 mode
 {
 	sys_fs.warning("sys_fs_open(path=%s, flags=%#o, fd=*0x%x, mode=%#o, arg=*0x%x, size=0x%llx)", path, flags, fd, mode, arg, size);
 
-	if (!path[0])
-	{
-		return CELL_EINVAL;
-	}
+	if (!path)
+		return CELL_EFAULT;
+
+	if (!*path.get_ptr())
```

Comment:
And what you didn't like in path[0] access method?

Reply:
I had to choose between `!path[0]` and `!*path.get_ptr()`. Both were already defined in the file to check for an empty string and my personal preference was to choose `!*path.get_ptr()`. I applied the same code across the file for consistency.
Is there a reason to use `!path[0]` instead of `!*path.get_ptr()`? Which one is better?

Reply:
Not to mention `!*path` will work equally, making get_ptr totally bloating. Indexing indicates clearly an access to an array of characters (which the string is).

Reply:
Understood, I will try to keep that in mind. I pushed the changes to use `!path[0]`.